tests/unit/ops/test_ops_schema.py

```diff
@@ -60,3 +62,195 @@ def test_schema_out(tags, properties, selection, op):
     not_used = [col for col in all_cols if col not in selector.names]
     for col_name in not_used:
         assert col_name not in new_schema.column_schemas
+
+
+@pytest.mark.parametrize("properties", [{"p1": "1"}])
+@pytest.mark.parametrize("tags", [["TAG1", "TAG2"]])
+@pytest.mark.parametrize(
+    "op_routine",
+    [
+        [ops.Categorify()],
+        [ops.Clip(min_value=10), ops.Categorify()],
+        [ops.Categorify(), ops.Rename(postfix="_test")],
+        [ops.Clip(min_value=10), ops.Categorify(), ops.Rename(postfix="_test")],
+    ],
+)
+def test_categorify_schema_properties(properties, tags, op_routine):
+    # Create columnSchemas
+    column_schemas = []
+    all_cols = []
+    for x in range(5):
+        all_cols.append(str(x))
+        column_schemas.append(ColumnSchema(str(x), tags=tags, properties=properties))
+
+    # Turn to Schema
+    schema = Schema(column_schemas)
+    df_dict = {}
+    num_rows = 10000
+    for column_name in schema.column_names:
+        df_dict[column_name] = np.random.randint(1, 1000, num_rows)
+
+    df = dispatch._make_df(df_dict)
+    dataset = nvt.Dataset(df)
+    test_node = ColumnSelector(schema.column_names) >> op_routine[0]
+    for op in op_routine[1:]:
+        test_node = test_node >> op
+    processor = nvt.Workflow(test_node)
+    processor.fit(dataset)
+    new_gdf = processor.transform(dataset).to_ddf().compute()
+    workflow_schema_out = processor.output_node.output_schema
+    for column_name in workflow_schema_out.column_names:
+        schema1 = workflow_schema_out.column_schemas[column_name]
+        assert "domain" in schema1.properties
+        embeddings_info = schema1.properties["domain"]
+        assert isinstance(embeddings_info, tuple)
+        # should always exist, represents unkown
+        assert embeddings_info[0] == 0
+        assert embeddings_info[1] == new_gdf[column_name].max() + 1
+    workflow_schema_out = {}
+    df_dict = {}
+    schema = {}
+
+
+@pytest.mark.parametrize("properties", [{}])
+@pytest.mark.parametrize("tags", [[]])
+@pytest.mark.parametrize(
+    "op_routine",
+    [
+        [ops.Categorify()],
+        [ops.Clip(min_value=10), ops.Categorify()],
+        [ops.Categorify(), ops.Rename(postfix="_test")],
+        [ops.Clip(min_value=10), ops.Categorify(), ops.Rename(postfix="_test")],
+    ],
+)
+def test_categorify_schema_properties_blank(properties, tags, op_routine):
```

Comment:
All subsequent tests are the same as above just with different parameters for tags and properties going into the workflow. If you try to collapse this down the way ti should be... (as a second entry to parameters in the first call, you will see that the first test of each "op_routine" will pass while all subsequent tests will fail. This is why this has been broken up. It's very ugly and hurts the eyes, but I feel the different behaviors need to be explicitly verified.

Reply:
Yeah, unfortunate it doesn't work the way we'd like here. Since the code of the tests is mostly the same, maybe we could extract a (non-test) helper method for the body of the test? Like most of the stuff above the assertions or something along those lines?